File: snippets/typescript/overwrite_payload.ts
import { QdrantClient } from "@qdrant/js-client-rest";

const client = new QdrantClient({ host: "localhost", port: 6333 });

client.overwritePayload("global_patient_data", {
  payload: {
    is_smoker: false,
    full_name: "Hiroshi Tanaka",
    country: "Japan",
    age: 58,
    gender: "Male",
    height: 175.3,
    date_of_birth: "1966-05-21",
    address: "1-2-3 Shibuya, Tokyo, 150-0002",
    notes:
      "Patient has a family history of diabetes. Regular check-ups recommended.",
  },
  points: [0],
});

File: snippets/go/overwrite_payload.go
package client

import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

func overwritePayload() {
	client, err := qdrant.NewClient(&qdrant.Config{
		Host: "localhost",
		Port: 6334,
	})
	if err != nil {
		panic(err)
	}

	client.OverwritePayload(context.Background(), &qdrant.SetPayloadPoints{
		CollectionName: "{collection_name}",
		Payload: qdrant.NewValueMap(map[string]any{
			"property1": "string",
			"property2": "string",
		}),
		PointsSelector: qdrant.NewPointsSelector(qdrant.NewIDNum(0), qdrant.NewIDNum(3), qdrant.NewIDNum(10)),
	})
}
